src/Sentry/Platforms/Cocoa/RuntimeMarshalManagedExceptionIntegration.cs

```diff
@@ -0,0 +1,45 @@
+using ObjCRuntime;
+using Sentry.Extensibility;
+using Sentry.Integrations;
+
+namespace Sentry.Cocoa;
+
+/// <summary>
+/// When AOT Compiling iOS applications, the AppDomain UnhandledExceptionHandler doesn't fire. So instead we intercept
+/// the Runtime.RuntimeMarshalManagedException event.
+/// </summary>
+internal class RuntimeMarshalManagedExceptionIntegration : ISdkIntegration
+{
+    private readonly IRuntime _runtime;
+    private IHub? _hub;
+    private SentryOptions? _options;
+
+    internal RuntimeMarshalManagedExceptionIntegration(IRuntime? runtime = null)
+        => _runtime = runtime ?? RuntimeAdapter.Instance;
+
+    public void Register(IHub hub, SentryOptions options)
+    {
+        _hub = hub;
+        _options = options;
+        _runtime.MarshalManagedException += Handle;
+    }
+
+    // Internal for testability
+    [SecurityCritical]
+    internal void Handle(object sender, MarshalManagedExceptionEventArgs e)
+    {
+        _options?.LogDebug("Runtime Marshal Managed Exception");
+
+        if (e.Exception is { } ex)
```

Comment:
`e` has: https://learn.microsoft.com/en-us/dotnet/api/objcruntime.marshalmanagedexceptionmode?view=xamarin-ios-sdk-12

We should add that detail as it can be useful context